docs/modules/ROOT/pages/reference/extensions/salesforce.adoc

```diff
@@ -84,6 +84,33 @@ To generate Salesforce DTOs for your project, use the `salesforce-maven-plugin`.
 </plugin>
 ----
 
+[id="extensions-salesforce-usage-native-mode-support-for-pub-sub-api-with-pojo-pubsubdeserializetype"]
+=== Native mode support for Pub / Sub API with POJO `pubSubDeserializeType`
+
+When using the Camel Salesforce Pub / Sub API and `pubSubDeserializeType` is configured as `POJO`, you must register any classes configured on the `pubSubPojoClass` option for reflection.
+
+For example, given the following route.
+
+[source,java]
+----
+from("salesforce:pubSubSubscribe:/event/TestEvent__e?pubSubDeserializeType=POJO&pubSubPojoClass=org.foo.TestEvent")
+    .log("Received Salesforce POJO topic message: ${body}");
+----
+
```

Comment:
Non blocking, maybe we could point to the cq doc section describing how to register to native.

Reply:
Latest commit adds a link to the native guide reflection section.